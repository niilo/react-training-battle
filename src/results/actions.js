import { FIGHT, WINNER_FOUND, ERROR } from "./actionTypes";
import { battle } from "../util/api";

export const fight = () => ({
  type: FIGHT
});

export const winnerFound = results => ({
  type: WINNER_FOUND,
  payload: results
});

export const error = error => ({
  type: ERROR,
  payload: error
});

export const doFight = () => (dispatch, getState) => {
  console.log("fight");
  const state = getState();
  const playerOneName = state.battle.playerOne.name;
  const playerTwoName = state.battle.playerTwo.name;
  if (playerOneName === null || playerTwoName === null) {
    return dispatch(error("failed."));
  }

  battle([playerOneName, playerTwoName]).then(results => {
    if (results === null) {
      return dispatch(error("failed."));
    }
    return dispatch(winnerFound(results));
    /*this.setState(() => {
      return {
        error: null,
        winner: results[0],
        loser: results[1],
        loading: false
      };
    });*/
  });
};
