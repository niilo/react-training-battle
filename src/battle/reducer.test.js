import * as a from "./actions";
import reducer from "./reducer";

const player1 = {
  profile: {
    name: "react",
    followers: 1,
    following: 1,
    public_repos: 2,
    stargazers_count: 73600,
    avatar_url: "https://avatars3.githubusercontent.com/u/69631?v=4"
  }
};

const player2 = {
  profile: {
    name: "react2",
    followers: 1,
    following: 1,
    public_repos: 2,
    stargazers_count: 73600,
    avatar_url: "https://avatars3.githubusercontent.com/u/69631?v=4"
  }
};

it("renders default correctly", () => {
  const state = reducer(undefined, a.setFirstPlayer(player1));
  const result = reducer(state, a.setSecondPlayer(player2));
  expect(result).toMatchSnapshot();
});
