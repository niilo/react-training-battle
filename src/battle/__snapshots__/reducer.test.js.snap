// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders default correctly 1`] = `
Object {
  "first": Object {
    "profile": Object {
      "avatar_url": "https://avatars3.githubusercontent.com/u/69631?v=4",
      "followers": 1,
      "following": 1,
      "name": "react",
      "public_repos": 2,
      "stargazers_count": 73600,
    },
  },
  "second": Object {
    "profile": Object {
      "avatar_url": "https://avatars3.githubusercontent.com/u/69631?v=4",
      "followers": 1,
      "following": 1,
      "name": "react2",
      "public_repos": 2,
      "stargazers_count": 73600,
    },
  },
}
`;
